Framework/DataHandling/src/LoadNexusProcessed2.cpp

```diff
@@ -163,47 +173,57 @@ void LoadNexusProcessed2::extractMappingInfoNew(const Mantid::NeXus::NXEntry &mt
  * Attempt to load nexus geometry. Should fail without exception if not
  * possible.
  *
- * Caveats are:
- * 1. Only works for input files where there is a single NXEntry. Does nothing
- * otherwise.
- * 2. Is only applied after attempted instrument loading in the legacy fashion
+ * Caveat is:
+ *   Is only applied after attempted instrument loading in the legacy fashion
  * that happens as part of loadEntry. So you will still get warning+error
  * messages from that even if this succeeds
  *
  * @param ws : Input workspace onto which instrument will get attached
- * @param nWorkspaceEntries : number of entries
+ * @param entryNumber: number of the NXentry for the parent group: used to construct the group's name
  * @param logger : to write to
- * @param filename : filename to load from.
+ * @param filePath : filename to load from.
  * @return true if successful
  */
-bool LoadNexusProcessed2::loadNexusGeometry(API::Workspace &ws, const int nWorkspaceEntries, Kernel::Logger &logger,
-                                            const std::string &filename) {
-  if (m_instrumentLayout == InstrumentLayout::NexusFormat && nWorkspaceEntries == 1) {
+bool LoadNexusProcessed2::loadNexusGeometry(API::Workspace &ws, size_t entryNumber, Kernel::Logger &logger,
+                                            const std::string &filePath) {
+  if (m_instrumentLayout == InstrumentLayout::NexusFormat) {
     if (auto *matrixWs = dynamic_cast<API::MatrixWorkspace *>(&ws)) {
       try {
         using namespace Mantid::NexusGeometry;
+
+        std::ostringstream parent_;
+        parent_ << "mantid_workspace_" << entryNumber;
```

Comment:
Should "mantid_workspace_" be a const somewhere (question scoped at the greater codebase)?  A [code search ](https://github.com/search?q=repo%3Amantidproject%2Fmantid%20mantid_workspace_&type=code)reveals that this is normally just accessed via a plain string, but I wasnt quite sure what the significance of this was when first looking at it, but it seems to be some standard expected name?

Reply:
I'd hesitate to make it a const.  That kind of change would have too wide of a scope over the entire Mantid framework.
You're correct that its the standard expected name.  Another name that's sometimes used is `raw_data_n`, but that's a special case.